core-bundle/contao/modules/ModuleSearch.php

```diff
@@ -131,7 +131,7 @@ protected function compile()
 				foreach ($GLOBALS['TL_HOOKS']['customizeSearch'] as $callback)
 				{
 					$this->import($callback[0]);
-					$this->{$callback[0]}->{$callback[1]}($arrPages, $strKeywords, $strQueryType, $blnFuzzy, $this);
+					$this->{$callback[0]}->{$callback[1]}($arrPages, $strKeywords, $strQueryType, $blnFuzzy, $this, $objResult, $arrResult);
```

Comment:
```suggestion
					$objResult = $this->{$callback[0]}->{$callback[1]}($arrPages, $strKeywords, $strQueryType, $blnFuzzy, $this);
```
I think we should use the return value of the hook instead here (and break the loop at the first hook that returns an object). Would this also work for you?